TicTacToe/CLI.cs

```diff
@@ -32,13 +32,14 @@ public string[] GetPlayerNames(Game game)
             return names;
         }
 
-        public int GetPlayerMove(Player player)
+        public int GetPlayerMove(Player player, int boardSize)
```

Comment:
By passing in the entire Player object into this method, you are allowing this class to "know" something about the inner workings of Player - i.e, that it has a property 'Name'.

Try to think of params as [data in, data out:](https://blog.jessitron.com/2015/08/03/data-in-data-out/)

> The output of the function is entirely determined by the values of its input. The function is isolated from the world around it.

Since you are only using Player inside of line 38, refactoring the method to accept `(String playerName, int boardSize)` would adhere to the data in, data out methodology. :)

Reply:
That's really helpful to think about, thanks – I will refactor with that in mind (also I'll see if there are any other places I can apply that data in/out paradigm)